poc/vdaf_prio3.sage

```diff
@@ -222,79 +227,96 @@ class Prio3(Vdaf):
         # oracle anyway, in order to apply the Fiat-Shamir heuristic.
         return Prio3.Prg.derive_seed(
             zeros(Prio3.Prg.SEED_SIZE),
-            dst + byte(255) + concat(k_joint_rand_parts))
+            dst + DST_JOINT_RAND_SEED + concat(k_joint_rand_parts))
 
     @classmethod
     def encode_leader_share(Prio3,
-                            input_share,
+                            measurement_share,
                             proof_share,
-                            k_blind,
-                            k_hint):
+                            k_blind):
         encoded = Bytes()
-        encoded += Prio3.Flp.Field.encode_vec(input_share)
+        encoded += Prio3.Flp.Field.encode_vec(measurement_share)
         encoded += Prio3.Flp.Field.encode_vec(proof_share)
         if Prio3.Flp.JOINT_RAND_LEN > 0:
             encoded += k_blind
-            encoded += concat(k_hint)
         return encoded
 
     @classmethod
     def decode_leader_share(Prio3, encoded):
         l = Prio3.Flp.Field.ENCODED_SIZE * Prio3.Flp.INPUT_LEN
-        encoded_input_share, encoded = encoded[:l], encoded[l:]
-        input_share = Prio3.Flp.Field.decode_vec(encoded_input_share)
+        encoded_measurement_share, encoded = encoded[:l], encoded[l:]
+        measurement_share = Prio3.Flp.Field.decode_vec(encoded_measurement_share)
         l = Prio3.Flp.Field.ENCODED_SIZE * Prio3.Flp.PROOF_LEN
         encoded_proof_share, encoded = encoded[:l], encoded[l:]
         proof_share = Prio3.Flp.Field.decode_vec(encoded_proof_share)
         l = Prio3.Prg.SEED_SIZE
-        k_blind, k_hint = None, None
-        if Prio3.Flp.JOINT_RAND_LEN > 0:
-            k_blind, encoded = encoded[:l], encoded[l:]
-            k_hint = []
-            for _ in range(Prio3.SHARES-1):
-                k_joint_rand_part, encoded = encoded[:l], encoded[l:]
-                k_hint.append(k_joint_rand_part)
+        if Prio3.Flp.JOINT_RAND_LEN <= 0:
```

Comment:
While technically correct, this code makes it look as if `-23` might be a valid value for `JOINT_RAND_LEN`. (Similarly below.)
```suggestion
        if Prio3.Flp.JOINT_RAND_LEN == 0:
```

Reply:
Funnily enough, i started with == and then changed it, in the hopes that if -23 showed up as an invalid value it would cause an ERR_DECODE.